### DISCUSSION TITLE
Facing issue to collect frequently changing datapoints using async gauge.

### DISCUSSION THREAD

**Discussion by ychaydhari** [open-telemetry/opentelemetry-java-instrumentation#7177] [Q&A]
Hi,
We have a few data points which vary very frequently within 1-2 secs. I'm using guageBuilder to collect this metric. but few data points are not able to capture. 

private final Meter meter =
            GlobalOpenTelemetry.meterBuilder("com.rs.mv.cm.stats")
                    .build();

meter.gaugeBuilder(OtelConstants.TOTAL_INUSE_CONNCTIONS)
                .setDescription("Total number of in use connections.")
                .setUnit("unit")
                .buildWithCallback(
                        r -> {
                                r.record(account.getConnectionsInUse());
                            }
                        });

Can someone have any other way to capture this type of data point? which are very frequent within 1-2 secs?

**Comment by trask** [comment]
Can you use a histogram?

**Reply by ychaydhari** [reply]
Hi @trask, Do we support synchronous gauges?  using histogram not getting proper values.

**Reply by mateuszrzeszutek** [reply]
Hey @ychaydhari ,
Not yet - there's an open spec issue about sync gauges (https://github.com/open-telemetry/opentelemetry-specification/issues/2318).
